### PR TITLE
Toplevelmenu v5

### PR DESCRIPTION
**Pull request by AndrewKhukhrianskyi**
1. Test case fixing in the Toplevelmenu Tab
2. Add new test case in Search Tab
3. Create new test cases in the Items Tab

### CODE REVIEW

File: pages/items_page.rb [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
+    browser.find_element(xpath:'//p[@id="quantity_wanted"]/input[@name ="qty"]')
+  end
+  
+  def quantity_enter(data)
+	quantity.clear
+	quantity.send_keys(data)
+  end
+  
+  def asc_btn
+	browser.find_element(xpath:'//a[@class="btn btn-default button-plus product_quantity_up"]')
```
Comment (olichka12) [inline pages/items_page.rb:14]: Can use just (class: 'product_quantity_up') the same for product_quantity_down

File: pages/tplvlmenu_tab.rb [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
   end
 
   def tshirts
-    browser.find_element(name: 'T-shirts')
+    browser.find_element(xpath: '//ul[@class="sf-menu clearfix menu-content sf-js-enabled sf-arrows"]/li/a[@title="T-shirts"]')
   end
 
   def tplm_bar
-    browser.find_element(class: 'sf-menu clearfix menu-content sf-js-enabled sf-arrows')
+    browser.find_element(xpath: '//ul[@class="sf-menu clearfix menu-content sf-js-enabled sf-arrows"]')
```
Comment (olichka12) [inline pages/tplvlmenu_tab.rb:17]: Please use class selector. This two rows mean the same

File: spec/items_spec.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  
+  context 'when valid and invalid credentials' do
+  
+	it 'verifies that quantities field is displayed'
+		expect(items_page.quantity.displayed?) to be(true)
+	end
+	
+	# ?
+    it 'verifies user can type numbers in the quantities field' do
+		data = 5
```
Comment (olichka12) [inline spec/items_spec.rb:48]: Move 'data' out of test.
And please correct alignment

File: pages/items_page.rb [hunk 4]
```diff
@@ -1,5 +1,18 @@
 class ItemsPage < BasePage
   def quantity
-    browser.find_element(id: 'quantity_wanted')
+    browser.find_element(xpath:'//p[@id="quantity_wanted"]/input[@name ="qty"]')
```
Comment (olichka12) [inline pages/items_page.rb:5]: Don't need this changes with xpath

File: pages/tplvlmenu_tab.rb [hunk 5]
```diff
@@ -1,17 +1,17 @@
 class TPLMTab < BasePage
   def women
-    browser.find_element(name: 'Women')
+    browser.find_element(xpath: '//ul[@class="sf-menu clearfix menu-content sf-js-enabled sf-arrows"]/li/a[@title="Women"]')
```
Comment (olichka12) [inline pages/tplvlmenu_tab.rb:5]: Seems with name select should be work.
Please check it

File: spec/top_level_menu_spec.rb [hunk 6]
```diff
@@ -26,5 +26,26 @@
     it 'verify that all tab with buttons is displayed' do
       expect(tplevelmenu_tab.tplm_bar.displayed?).to be(true)
     end
+	
+	it 'verify that after clickng on the women button women page is opened'
+		current_url = browser.current_url
+		tplevelmenu_tab.women.click
+		search_page.wait_for{ current_url != browser.current_url }
+		expect(tplevelmenu_tab.url_include?('http://automationpractice.com/index.php?id_category=3&controller=category').on page? (true)
```
Comment (olichka12) [inline spec/top_level_menu_spec.rb:35]: Can be constant like WOMEN_PAGE = "#{HOME_PAGE}/index.php?id_category=3&controller=category", the same for dress and t-shirts  
expect(current_url).to eq (WOMEN_PAGE)
Would be great to check if exist some context of this page (can be element of this page)

Also you can use shared_examples for this tests

File: pages/tplvlmenu_tab.rb [hunk 7]
```diff
@@ -1,17 +1,37 @@
...
-    browser.find_element(class: 'sf-menu clearfix menu-content sf-js-enabled sf-arrows')
+    browser.find_element(xpath: '//ul[@class="sf-menu clearfix menu-content sf-js-enabled sf-arrows"]')
   end
+  
+  def home_icon
+	browser.find_element(class: 'home')
+  end
+  
+  def dresses_label
+	browser.find_element(xpath: '//div[@class="breadcrumb clearfix"]/span["Women"]')
```
Comment (AndriiMish) [inline pages/tplvlmenu_tab.rb:25]: add label by name method
def label_by_name(name)
  browser.find_element(xpath: "//div[@class="breadcrumb clearfix"]/span[#{name}]")
end
will remove duplication

File: spec/search_spec.rb [hunk 8]
```diff
@@ -18,13 +34,25 @@
...
       expect(search_page.url_include?(data)).to be(true)
     end
+	
+	it 'verifies that user request which consist unexisted data can not find anything' do
+      data = 'asdsadsada'
+      current_url = browser.current_url
+      search_page.search(data)
+      search_page.wait_for{ current_url != browser.current_url }
+      expect(search_page.results_label) == '0 results have been found.'
+    end
```
Comment (AndriiMish) [inline spec/search_spec.rb:55]: seems few examples are same, the only changing thing is data. consider using shared examples or arrays\

File: spec/top_level_menu_spec.rb [hunk 9]
```diff
@@ -19,12 +23,76 @@
...
+		expect(tplevelmenu_tab.home_icon.displayed?).to be (false)
+	end
+	
+	it 'verify that after clicking on the home icon user will be redirected to the main page'
+		current_url = browser.current_url
+		tplevelmenu_tab.women.click
+		tplevelmenu_tab.wait_for{ current_url != browser.current_url }
+		tplevelmenu_tab.home_icon.click
+		expect(tplevelmenu_tab.url_include?('http://automationpractice.com/index.php'))
+	end
```
Comment (AndriiMish) [inline spec/top_level_menu_spec.rb:94]: lots of duplication in examples

**Comment by AndriiMish** [comment]
Addressed in https://github.com/AndriiMish/TAQC/pull/27

**Pull request closed without merging**